### ISSUE TITLE
Zone01 crashes if two nameservers serve different SOA serials

### ISSUE THREAD

**Issue by marc-vanderwal** [zonemaster/zonemaster-engine#1175]
Consider the following test zone, served by two nameservers but with different SOA serials:

```zone
$TTL 3600
$ORIGIN mname-not-master.zone01.xa.

@                       IN  SOA  (
                                   ns1.mname-not-master.zone01.xa.
                                   root.nic.test.
                                   2022122300 ; ns1 serves this serial, ns2 a slightly higher one
                                   86400
                                   14400
                                   3600000
                                   3600
                                 )

                            NS   ns1.mname-not-master.zone01.xa.
                            NS   ns2.mname-not-master.zone01.xa.

ns1                         A    127.0.1.1
ns2                         A    127.0.1.2
```

When running `zonemaster-cli` on such a broken zone, the test unfortunately crashes with a fatal error:

```
$ zonemaster-cli --test zone/zone01 --hints local.hints mname-not-master.zone01.xa
Seconds Level     Message
======= ========= =======
   0.04 CRITICAL  Fatal error in Zone: Undefined subroutine &Zonemaster::Engine::Test::Zone::max called at [redacted]/Zonemaster/Engine/Test/Zone.pm line 561.
```

**Comment by matsduf** [comment]
> When running `zonemaster-cli` on such a broken zone, the test unfortunately crashes with a fatal error:

I would not even call it a broken zone. It is normal for two servers to serve different SOA serial just after an update of the zone. This a bad bug.

**Comment by mdavids** [comment]
I encountered is as well:

https://zonemaster.net/en/result/64d899fe370d3c4f

**Comment by tgreenx** [comment]
> I encountered it as well:
> 
> https://zonemaster.net/en/result/64d899fe370d3c4f

Hi @mdavids,

This will be fixed by #1178 in next fix release planned for 23/01/2023. But if you'd rather have a fix right now you can already look at the changes in the PR and implement them yourself.

**Comment by tgreenx** [comment]
Solved by #1178
